### PR TITLE
feat: pass session on object instantiation

### PR DESCRIPTION
**Pull request by Xstoudi**

### CODE REVIEW

File: src/Model/Model.ts [hunk 1]
```diff
@@ -97,12 +99,14 @@ export class Model {
   protected $originalData: any;
   protected $currentData: any;
   protected $isDeleted: boolean;
+  protected $options: any;
```
Comment (targos) [inline src/Model/Model.ts:102]:
```suggestion
  protected $options: IModelOptions;
```

**Comment by Xstoudi** [comment]
I think the typical usecase would eventually appear if we support this syntax in the futur:
```js
const user = new User()
user.username = 'root'
user.password = 'root'
await user.save({ session })
```

**Comment by targos** [comment]
I don't see a usecase either, but is there a good reason for explicitly disallowing it?

**Comment by stropitek** [comment]
> I don't see a usecase either, but is there a good reason for explicitly disallowing it?

For me it was about not misleading the user. If the option is there it might be interpreted as "you have to pass a session in order to use the operation in a transaction". But I'm ok either way, it will also keep the typing simpler if we still allow it

**Pull request merged**